integration/test_ml.py

```diff
@@ -0,0 +1,380 @@
+
+
+# Test tensor flow conversion functions if tensor flow is enabled.
+#'pip install tensorflow' in the environment if you want _TF_ENABLED = True
+#'pip install tensorflow==2.0.0b' for version 2 etc.
+
+
+SAVED_MODEL_DIR = '/tmp/saved_model/1'
+
+
+def _clean_up_tmp_directory():
+    if os.path.exists(SAVED_MODEL_DIR):
+        shutil.rmtree(SAVED_MODEL_DIR)
+
+
+@pytest.fixture
+def keras_model():
+    assert _TF_ENABLED
+    x_array = [-1, 0, 1, 2, 3, 4]
+    y_array = [-3, -1, 1, 3, 5, 7]
+    model = tf.keras.models.Sequential(
+        [tf.keras.layers.Dense(units=1, input_shape=[1])])
+    model.compile(optimizer='sgd', loss='mean_squared_error')
+    model.fit(x_array, y_array, epochs=3)
+    return model
+
+
+@pytest.fixture
+def saved_model_dir(keras_model):
+    assert _TF_ENABLED
+    # different versions have different model conversion capability
+    # pick something that works for each version
+    save_dir = SAVED_MODEL_DIR
+    _clean_up_tmp_directory() # previous failures may leave files
+    if tf.version.VERSION.startswith('1.'):
+        tf.reset_default_graph()
+        x_var = tf.placeholder(tf.float32, (None, 3), name="x")
+        y_var = tf.multiply(x_var, x_var, name="y")
+        with tf.Session() as sess:
+            tf.saved_model.simple_save(sess, save_dir, {"x": x_var}, {"y": y_var})
+    else:
+        # If it's not version 1.x or version 2.x we need to update the test.
+        assert tf.version.VERSION.startswith('2.')
+        tf.saved_model.save(keras_model, save_dir)
+    yield save_dir
+    _clean_up_tmp_directory()
+
+
+@unittest.skipUnless(_TF_ENABLED, 'Tensor flow is required for this test.')
+def test_from_keras_model(keras_model):
+    source = ml.TFLiteGCSModelSource.from_keras_model(keras_model, 'model2.tflite')
+    assert re.search(
+        '^gs://.*/Firebase/ML/Models/model2.tflite$',
+        source.gcs_tflite_uri) is not None
+
+    # Validate the conversion by creating a model
+    try:
+        model_format = ml.TFLiteFormat(model_source=source)
+        model = ml.Model(display_name="KerasModel1", model_format=model_format)
+        created_model = ml.create_model(model)
+        assert created_model.model_id is not None
+        assert created_model.validation_error is None
+    finally:
+        _clean_up_model(created_model)
+
+
+@unittest.skipUnless(_TF_ENABLED, 'Tensor flow is required for this test.')
+def test_from_saved_model(saved_model_dir):
+    # Test the conversion helper
+    source = ml.TFLiteGCSModelSource.from_saved_model(saved_model_dir, 'model3.tflite')
+    assert re.search(
+        '^gs://.*/Firebase/ML/Models/model3.tflite$',
+        source.gcs_tflite_uri) is not None
+
+    # Validate the conversion by creating a model
+    try:
+        model_format = ml.TFLiteFormat(model_source=source)
+        model = ml.Model(display_name="SavedModel1", model_format=model_format)
+        created_model = ml.create_model(model)
```

Comment:
Move the first 3 lines out of the `try` block. Makes the semantics easier to understand.